Bullet/Bullet.gd

```diff
@@ -17,5 +21,21 @@ func _physics_process(delta: float) -> void:
 	var collision := move_and_collide(velocity * delta)
 	if collision:
 		if collision.collider.is_in_group("enemies") or collision.collider.is_in_group("crates"):
-			collision.collider.on_damaged(_DAMAGE)
+			var damage = _get_damage()
+			collision.collider.on_damaged(damage)
+			var effectiveness = (damage - (_MEAN_DAMAGE - _SPREAD / 2)) / _SPREAD
+			_show_damage_number(damage, effectiveness)
 			queue_free()
+
+func _get_damage() -> int:
+	var lower_bound = _MEAN_DAMAGE - _SPREAD / 2
+	var upper_bound = _MEAN_DAMAGE + _SPREAD / 2
+	var damage_float := randf() * _SPREAD + _MEAN_DAMAGE - _SPREAD / 2
```

Comment:
`_MEAN_DAMAGE + (randf() - 1/2) * _SPREAD` would this be more readable/intuitive?

Reply:
Oh yes. Actually I wanted to use `rand_range` but seems like i needed to use a RandomNumberGenerator. So the lower and upper bound values weren't actually used...

Reply:
oh I didnt even realize they weren't used lol